paddle/fluid/imperative/prepared_operator.h

```diff
@@ -39,24 +42,120 @@ namespace imperative {
 
 const framework::Tensor* GetTensorFromVar(const framework::Variable& var);
 
+template <typename VarType>
+static void PrepareGradVarDataType(const std::shared_ptr<VarType>& var);
```

Comment:
PrepareGradVarDataType  is a little bit confusing, better SaveForwardDataTypeInGradVar or other?

Reply:
done, change to `SetForwardDataTypeOfGradVar`